samples/browser/src/main/java/org/mozilla/samples/browser/addons/AddOnsFragment.kt

```diff
@@ -152,4 +162,44 @@ class AddOnsFragment : Fragment(), View.OnClickListener {
             }
         }
     }
+
+    private fun isAlreadyADialogCreated(): Boolean {
+        return findPreviousDialogFragment() != null
+    }
+
+    private fun findPreviousDialogFragment(): PermissionsDialogFragment? {
+        return fragmentManager?.findFragmentByTag(PERMISSIONS_DIALOG_FRAGMENT_TAG) as? PermissionsDialogFragment
+    }
+
+    private fun showPermissionDialog(addOn: AddOn) {
```

Comment:
We're going to get the list of permissions from GeckoView passed to a callback during installation. However, since these permissions are the same and we already have a translated list, I think it makes sense to display them to the user right away. If approved, we start installation and simply approve (return ALLOWED in the callback). We need the callback in GV anyway in case permissions change and need to be re-approved during an update.

Reply:
Let's talk about updates separately though...